src/main/java/frc/team2412/robot/commands/indexer/IndexBitmapCommand.java

```diff
@@ -0,0 +1,168 @@
+      final IndexerSensorSubsystem indexerSensorSubsystem) {
+    m_indexerMotorSubsystem = indexerMotorSubsystem;
+    m_indexerSensorSubsystem = indexerSensorSubsystem;
+    addRequirements(indexerMotorSubsystem);
+  }
+
+  @Override
+  public void execute() {
+    IntakeDirection intakeDirection = RobotState.getintakeDirection();
+    final boolean intakeOn = (intakeDirection != IntakeDirection.NONE);
+    if (intakeDirection == IntakeDirection.NONE) {
+      if (s_lastIntakeDirection == IntakeDirection.BACK) {
+        intakeDirection = IntakeDirection.BACK;
+      } else {
+        intakeDirection = IntakeDirection.FRONT;
+      }
+    } else {
+      s_lastIntakeDirection = intakeDirection;
+    }
+
+    int sensorBitmap = 0;
+    switch (intakeDirection) {
+    case BOTH:
+    case FRONT:
+      sensorBitmap = (m_indexerSensorSubsystem.getSensorBitmapFrontLSB() & VALID_SENSOR_BITS);
+      break;
+    case BACK:
+      sensorBitmap = (m_indexerSensorSubsystem.getSensorBitmapBackLSB() & VALID_SENSOR_BITS);
+      break;
+    case NONE:
+      assert (false);
+      break;
+    }
+
+    IndexCommandEntry indexCommand = null;
+    for (final IndexCommandEntry entry : IndexCommandEntry.values()) {
+      if (entry.expectedBits == (sensorBitmap & entry.validBits)) {
+        indexCommand = entry;
+      }
+    }
+
+    if (indexCommand != null) {
+      final IndexDirection frontIndexDirection = indexCommand.getFrontIndexDirection(intakeOn);
+      final IndexDirection backIndexDirection = indexCommand.getBackIndexDirection(intakeOn);
+
+      switch (intakeDirection) {
+        case BOTH:
+        case FRONT:
+          m_indexerMotorSubsystem.setFrontMotor(getIndexerMotorSpeed(frontIndexDirection));
+          m_indexerMotorSubsystem.setBackMotor(getIndexerMotorSpeed(backIndexDirection));
+          break;
+        case BACK:
+          // Swap the front & back motor values since the IndexCommandEntry assumes intake from the front
+          m_indexerMotorSubsystem.setBackMotor(getIndexerMotorSpeed(frontIndexDirection));
+          m_indexerMotorSubsystem.setFrontMotor(getIndexerMotorSpeed(backIndexDirection));
+          break;
+        default:
+          m_indexerMotorSubsystem.setFrontMotor(IndexerMotorSubsystem.MOTOR_OFF_SPEED);
+          m_indexerMotorSubsystem.setBackMotor(IndexerMotorSubsystem.MOTOR_OFF_SPEED);
+          break;
+        }
+      } else {
+        m_indexerMotorSubsystem.setFrontMotor(IndexerMotorSubsystem.MOTOR_OFF_SPEED);
+        m_indexerMotorSubsystem.setBackMotor(IndexerMotorSubsystem.MOTOR_OFF_SPEED);
+    }
+  }
+
+  @Override
+  public boolean isFinished() {
+    return true;
+  }
+
+  private double getIndexerMotorSpeed(IndexDirection direction) {
+    switch (direction) {
+      case IN:
+        return IndexerMotorSubsystem.MOTOR_IN_SPEED;
+      case OUT:
+        return IndexerMotorSubsystem.MOTOR_OUT_SPEED;
+      case OFF:
+      default:
+        return IndexerMotorSubsystem.MOTOR_OFF_SPEED;
+    }
```

Comment:
Um, so we have constants in IndexerConstants, and adding extra will just make this more confusing

Reply:
Fixed